pkg/scheduler/actions/enqueue/enqueue.go

```diff
@@ -114,7 +118,23 @@ func (enqueue *Action) Execute(ssn *framework.Session) {
 			inqueue = true
 		} else {
 			minReq := api.NewResource(*job.PodGroup.Spec.MinResources)
+			if _, ok := resourceMap[job.Namespace]; !ok && options.ServerOpts.ConsiderResourceQuotaDuringEnqueue {
+				resource := ssn.NamespaceInfo[api.NamespaceName(job.Namespace)].GetResource()
+				if !reflect.DeepEqual(resource, api.EmptyResource()) {
+					resourceMap[job.Namespace] = resource
+				}
+			}
+			klog.V(3).Infof("Now start to do comparison, for job <%s/%s>, Enqueueable: %t, idle: %t", job.Namespace, job.Name, ssn.JobEnqueueable(job), minReq.LessEqual(idle))
 			if ssn.JobEnqueueable(job) && minReq.LessEqual(idle) {
+				if _, ok := resourceMap[job.Namespace]; ok {
+					klog.V(3).Infof("For job <%s/%s>, ResourceQuota: %t", job.Namespace, job.Name, minReq.LessEqual(resourceMap[job.Namespace]))
+					if minReq.LessEqual(resourceMap[job.Namespace]) {
+						resourceMap[job.Namespace].Sub(minReq)
```

Comment:
l think its better to put `idle.Sub(minReq); inqueue = true` here so that the following `else` clause can be removed :)

Reply:
Actually `idle.Sub(minReq)；inqueue = true`  doesn't belong to this `if` branch;

If we put it here, `idle.Sub` will not be calculated under circumstances when there is no resourceQuota in the namespace.